backends-common/postgres/src/main/java/org/apache/james/backends/postgres/quota/PostgresQuotaCurrentValueDAO.java

```diff
@@ -0,0 +1,164 @@
+import static org.jooq.impl.DSL.greatest;
+import static org.jooq.impl.DSL.zero;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+import org.apache.james.backends.postgres.utils.PostgresExecutor;
+import org.apache.james.core.quota.QuotaComponent;
+import org.apache.james.core.quota.QuotaCurrentValue;
+import org.apache.james.core.quota.QuotaType;
+import org.jooq.Record;
+
+import com.google.common.base.MoreObjects;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class PostgresQuotaCurrentValueDAO {
+    public static class QuotaKey {
+
+        public static QuotaKey of(QuotaComponent component, String identifier, QuotaType quotaType) {
+            return new QuotaKey(component, identifier, quotaType);
+        }
+
+        private final QuotaComponent quotaComponent;
+        private final String identifier;
+        private final QuotaType quotaType;
+
+        public QuotaComponent getQuotaComponent() {
+            return quotaComponent;
+        }
+
+        public String getIdentifier() {
+            return identifier;
+        }
+
+        public QuotaType getQuotaType() {
+            return quotaType;
+        }
+
+        private QuotaKey(QuotaComponent quotaComponent, String identifier, QuotaType quotaType) {
+            this.quotaComponent = quotaComponent;
+            this.identifier = identifier;
+            this.quotaType = quotaType;
+        }
+
+        @Override
+        public final int hashCode() {
+            return Objects.hash(quotaComponent, identifier, quotaType);
+        }
+
+        @Override
+        public final boolean equals(Object o) {
+            if (o instanceof QuotaKey) {
+                QuotaKey other = (QuotaKey) o;
+                return Objects.equals(quotaComponent, other.quotaComponent)
+                    && Objects.equals(identifier, other.identifier)
+                    && Objects.equals(quotaType, other.quotaType);
+            }
+            return false;
+        }
+
+        public String toString() {
+            return MoreObjects.toStringHelper(this)
+                .add("quotaComponent", quotaComponent)
+                .add("identifier", identifier)
+                .add("quotaType", quotaType)
+                .toString();
+        }
+    }
+
+    private final PostgresExecutor postgresExecutor;
+
+    public PostgresQuotaCurrentValueDAO(PostgresExecutor postgresExecutor) {
+        this.postgresExecutor = postgresExecutor;
+    }
+
+    public Mono<Void> increase(QuotaKey quotaKey, long amount) {
+        return postgresExecutor.executeVoid(dslContext -> Mono.from(dslContext.insertInto(TABLE_NAME)
+            .set(IDENTIFIER, quotaKey.getIdentifier())
+            .set(COMPONENT, quotaKey.getQuotaComponent().getValue())
+            .set(TYPE, quotaKey.getQuotaType().getValue())
+            .set(CURRENT_VALUE, amount)
+            .onConflict(IDENTIFIER, COMPONENT, TYPE)
```

Comment:
You can use the same way in my PR to avoid enumerating again all components of the primary key